Build failure on CentOS Stream 8

As first mentioned in https://github.com/lkrg-org/lkrg/pull/140#issuecomment-1012268148:

Somehow `build (quay.io/centos/centos:stream8` failed with:

```
make -C /lib/modules/4.18.0-358.el8.x86_64/build M=/__w/lkrg/lkrg modules
make[1]: Entering directory '/usr/src/kernels/4.18.0-358.el8.x86_64'
  CC [M]  /__w/lkrg/lkrg/src/modules/ksyms/p_resolve_ksym.o
  CC [M]  /__w/lkrg/lkrg/src/modules/hashing/p_lkrg_fast_hash.o
In file included from ./include/linux/kernel.h:11,
                 from /__w/lkrg/lkrg/src/modules/ksyms/../../p_lkrg_main.h:24,
                 from /__w/lkrg/lkrg/src/modules/ksyms/p_resolve_ksym.c:19:
/__w/lkrg/lkrg/src/modules/ksyms/../../modules/exploit_detection/p_exploit_detection.h: In function 'p_get_task_state':
/__w/lkrg/lkrg/src/modules/ksyms/../../modules/exploit_detection/p_exploit_detection.h:159:29: error: 'struct task_struct' has no member named 'state'; did you mean '__state'?
    return READ_ONCE(p_task->state);
                             ^~~~~
```

Has the rename of state to __state from 5.14+ been just back-ported to CentOS Stream 8? We need to confirm this and likely introduce a proper (exact starting revision) RHEL kernel version check into our `p_get_task_state` (treating those new RHEL kernels the same as 5.14+).

--- a/src/modules/exploit_detection/p_exploit_detection.h
+++ b/src/modules/exploit_detection/p_exploit_detection.h
@@ -151,7 +151,8 @@
 
 static inline unsigned int p_get_task_state(struct task_struct *p_task) {
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,14,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,14,0) || \
+  (defined(RHEL_RELEASE_CODE) && RHEL_RELEASE_CODE > RHEL_RELEASE_VERSION(8,5))
    return READ_ONCE(p_task->__state);
 #elif defined(ACCESS_ONCE) && !defined(READ_ONCE)
    return ACCESS_ONCE(p_task->state);

